lib/maths/CBoostedTreeImpl.cc

```diff
@@ -39,6 +41,9 @@ namespace {
 // quantiles anyway. So we amortise their compute cost w.r.t. training trees
 // by only refreshing once every MINIMUM_SPLIT_REFRESH_INTERVAL trees we add.
 const double MINIMUM_SPLIT_REFRESH_INTERVAL{3.0};
+const std::string HYPERPARAMETER_OPTIMIZATION_PHASE{"hyperparameter_optimization"};
+const std::string TRAINING_FINAL_TREE_PHASE{"training_final_tree"};
+const std::array<std::string, 1> REGRESSION_LOSSES{CMse::NAME};
```

Comment:
My inclination would be to add an `isRegression()` to the loss function hierarchy rather than rely on this hidden array: it means that when you add a loss function you know you have to define this quantity rather. As it is it would be easy to miss that you need to update this array.

Reply:
I would like to avoid working on the same files as you at the moment. I will put a TODO and change it in a follow-up PR, ok?

Reply:
Agreed. (In fact, I need to add an `isClassification` or equivalent for the PR which wires in multiclass, so we can wait for this to arrive.)